keystone/common/api_filtering.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2015 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import copy
+from keystone import config
+from keystone import exception
+from keystone.models import token_model
+from keystone.i18n import _LE
+from oslo_config import cfg
+from oslo_log import log
+from oslo_log import loggers
+from functools import wraps
+
+LOG = log.getLogger(__name__)
+
+
+def filter_projects(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+
+        res = f(*args, **kwargs)
+        is_admin_flag = _is_admin(args[0], args[1])
+
+        if not is_admin_flag:
+            if kwargs is not None and kwargs.get('project_id') is not None:
+                if (res['project']['name']
+                        in CONF.api_filtering.projects_to_filter):
+                    msg = ('Project \'%s\' does not exist.'
+                           % kwargs.get('project_id'))
+                    LOG.error(_LE(msg))
+                    raise exception.NotFound(msg)
+            else:
+                refs = [i for i in res['projects']]
+                for i in refs:
```

Comment:
It is generally better practice to use meaningful variable names as it makes it easier to read.  "i" has no context for a reader/reviewer.